src/freenet/client/filter/HTMLFilter.java

```diff
@@ -2616,7 +2619,11 @@ void processStyle(HTMLParseContext pc) {
 			String href = getHashString(h, "href");
 			if (href != null) {
 				href = HTMLDecoder.decode(href);
-				href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				if (isIcon) {
+					href = htmlSanitizeURI(href, type, null, null, pc.cb, pc, false);
+				} else {
+					href = htmlSanitizeURI(href, type, charset, maybecharset, pc.cb, pc, false);
+				}
```

Comment:
a) it didn’t occur to me to clobber variables. :)
b) now that I thought about it I still think it would be inferior to what I actually did.

If I would have written it the way you suggest it would be easy for the two resulting pieces of code (the if-block with the null-setting inside and the call to sanitizeURI) to be moved away from each other which would make the intent of not having a charset on an image — which is what this is about! — a lot less clearer. The way I did it it’s always clear that something different happens for an icon.